harmoni-web-api/src/main/java/org/harmoniapp/harmoniwebapi/services/MessageService.java

```diff
@@ -0,0 +1,92 @@
+package org.harmoniapp.harmoniwebapi.services;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.harmoniapp.harmonidata.entities.Message;
+import org.harmoniapp.harmonidata.entities.User;
+import org.harmoniapp.harmonidata.repositories.RepositoryCollector;
+import org.harmoniapp.harmoniwebapi.contracts.MessageDto;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@ComponentScan(basePackages = {"org.harmoniapp.harmonidata"})
+public class MessageService {
+    private final RepositoryCollector repositoryCollector;
+    private final SimpMessagingTemplate messagingTemplate;
+    private final TranslationService translationService;
+
+//    public List<MessageDto> getChatHistory(Long userId1, Long userId2) {
+//        List<Message> messages= repositoryCollector.getMessages().findChatHistory(userId1, userId2);
+//        return messages.stream()
+//                .map(MessageDto::fromEntity)
+//                .toList();
+//    }
+
+    public List<MessageDto> getChatHistory(Long userId1, Long userId2, boolean translate, String targetLanguage) {
+        List<Message> messages = repositoryCollector.getMessages().findChatHistory(userId1, userId2);
+
+        return messages.stream()
+                .map(message -> {
+                    String content = message.getContent();
+
+                    if (translate && targetLanguage != null && !targetLanguage.isEmpty()) {
+                        content = translationService.translate(message.getContent(), targetLanguage);
+                    }
```

Comment:
You can omit the `boolean translate` variable when checking whether you need to translate and use only the `targetLanguage` value.